### PR TITLE
feat(dlp): added a sample for dlp deidentify reidentify deterministic

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [dlp reidentify deterministic](https://cloud.google.com/dlp/docs/samples/dlp-reidentify-deterministic) with a unit test case.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid/reid_with_deterministic.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// The wrapped key is base64-encoded, but the library expects a binary
+	// string, so decode it here.
+	keyBytes, err := base64.StdEncoding.DecodeString(wrappedKey)
+	if err != nil {
+		fmt.Fprintf(w, "error %v", err)
```
Comment (tritone) [inline dlp/snippets/deid/reid_with_deterministic.go:56]: Remove print logging.

File: dlp/snippets/deid/reid_with_deterministic.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		return err
+	}
+
+	// create crypto deterministic config.
```
Comment (tritone) [inline dlp/snippets/deid/reid_with_deterministic.go:60]: nit: Capitalize all comments.

**Review by tritone — APPROVED** [review]
Overall lgtm, just two nits.

File: dlp/snippets/deid/reid_with_deterministic.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/reid_with_deterministic.go]: Done.

File: dlp/snippets/deid/reid_with_deterministic.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/reid_with_deterministic.go]: Done.

**Review by annadata — APPROVED** [review]

**Pull request merged**